phpBB/phpbb/language/loader/core_language_file_loader.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\language\loader;
+
+/**
+ * Language loader for phpBB's core language files
+ */
+class core_language_file_loader extends language_file_loader_base
+{
+	/**
+	 * @var string	Path to phpBB's root
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string	Custom language path
+	 */
+	protected $lang_dir;
+
+	/**
+	 * Constructor
+	 *
+	 * @param string	$phpbb_root_path	Path to phpBB's root
+	 */
+	public function __construct($phpbb_root_path)
+	{
+		$this->phpbb_root_path	= $phpbb_root_path;
+		$this->lang_dir			= false;
+	}
+
+	public function set_custom_language_path($lang_path)
+	{
+		$this->lang_dir = $lang_path;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function load($component, $locale, &$lang, &$help, $use_help = false)
+	{
+		if (!is_array($locale))
+		{
+			$this->locale = array($locale);
+		}
+		else
+		{
+			$this->locale = $locale;
+		}
+
+		// Determine path to language directory
+		$path = ($this->lang_dir !== false) ? $this->lang_dir : $this->phpbb_root_path . 'language/';
+
+		// Determine filename
+		if ($use_help && strpos($component, '/') !== false)
+		{
+			$filename = dirname($component) . '/help_' . basename($component);
```

Comment:
if we change all this, I think it's time to drop the help_ stuff.
I have a PR locally which turns the faq page to a normal language var based feature, so we don't need this hack anymore ;)